Whatsapp_Chat_Exporter/utility.py

```diff
@@ -328,7 +328,73 @@ def slugify(value, allow_unicode=False):
     return re.sub(r'[-\s]+', '-', value).strip('-_')
 
 
-class WhatsAppIdentifier(StrEnum):
+def find_contact_hash_ios(backup_location):
+    """
+    Find the contact hash for iOS
+    :param backup_location: The location of the backup (string)
+    :return: The contact hash (string)
+    """
+
+    manifest = os.path.join(backup_location, "Manifest.db")
+    if not os.path.isfile(manifest):
+        raise FileNotFoundError("Manifest.db not found")
+    import sqlite3
+
+    # we have to search for AppDomainGroup-group.net.whatsapp.WhatsApp.sharedMessage/Media/<Some arbitrary number>@s.whatsapp.net
+    # find a line containing AppDomainGroup-group.net.whatsapp.WhatsApp.sharedMessage/Media/
+    # SQLite format 3
+
+    # Connect to the SQLite database
+    conn = sqlite3.connect(manifest)
+    cursor = conn.cursor()
+    # Get all table names
+    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
+    tables = cursor.fetchall()
+
+    # Strings to search for
+    domain_search_string = "AppDomainGroup-group.net.whatsapp.WhatsApp.shared"
+    relative_path_search_string = "%.opus"  # Search for .opus at the end of the relative path
+    binary_plist_signature = b'62706c6973743030'  # Binary plist signature in hex, in text its bplist00
+
+    # Iterate over all tables and columns
```

Comment:
There should be no need to iterate all over the tables and columns? The `Files` table is what you want.

Reply:
Thanks! fixed